framerail/src/routes/[slug]/[...extra]/+server.ts

```diff
@@ -18,55 +25,88 @@ export async function POST(event) {
 
   let res: object = {}
 
-  if (extra.includes("edit")) {
-    /** Edit or create page. */
-    let comments = data.get("comments")?.toString() ?? ""
-    let wikitext = data.get("wikitext")?.toString()
-    let title = data.get("title")?.toString()
-    let altTitle = data.get("alt-title")?.toString()
-    let tagsStr = data.get("tags")?.toString().trim()
-    let tags: string[] = []
-    if (tagsStr?.length) tags = tagsStr.split(" ").filter((tag) => tag.length)
-
-    res = await page.pageEdit(
-      siteId,
-      pageId,
-      slug,
-      comments,
-      wikitext,
-      title,
-      altTitle,
-      tags
+  try {
+    if (extra.includes("edit")) {
+      /** Edit or create page. */
+      let comments = data.get("comments")?.toString() ?? ""
+      let wikitext = data.get("wikitext")?.toString()
+      let title = data.get("title")?.toString()
+      let altTitle = data.get("alt-title")?.toString()
```

Comment:
Does this also need `?? ""`?

Reply:
I didn't look at how deepwell implements page edit, but I am thinking more along the lines of "if the request didn't specify a new value for the field, leave it undefined and we can choose to carry it on from the previous revision", and comment is the thing that should not be carried over so it has a empty string default.

Reply:
Oh no, you're absolutely right, this should be left as-is.

The implementation is as you says, aside from the revision comments (which always must be populated) any body field can be omitted to say "leave it as the previous value". So if it's not specified it shouldn't be sent,